### ISSUE TITLE
Add start date and adjust activity dates to reflect start date

### ISSUE THREAD

**Issue by emmarichardson** [emeneo/moodle-local_course_templates#12]
When you restore a course through normal Moodle interface, activity dates will adjust to reflect the start date.  It would amazing if this functionality could be added to the course templates plugin...

**Comment by andrewscaya** [comment]
After analysis, this issue seems to be related to [MDL-41652](https://tracker.moodle.org/browse/MDL-41652).

I think this issue can be closed, because it is related to Moodle's internal mechanisms, and it is not directly related to this plugin's functionality.

@emmarichardson I recommend that this discussion be taken up with Moodle HQ.

@Flottertotte What do you think?

**Comment by Flottertotte** [comment]
@andrewscaya, yes, I think so, too. From what I read there it seems that part of the backup-restore functions already include that date feature, but not sure if the related moodle function which we are using includes it or not.

**Comment by andrewscaya** [comment]
@Flottertotte Yes, we use the standard duplicate_course method from the core_course_external class (it uses the backup/restore API under the hood).

**Comment by Flottertotte** [comment]
@andrewscaya yes, I know. But I am not sure if it is fixed there. See last post in here [MDL-41652](https://tracker.moodle.org/browse/MDL-41652) :

> Hi, this is still happening on Moodle 3.11dev+ 
If you do a manual backup and restore of a course, the dates roll up correctly relative to the new start of date of the course restored as they were relative to the original date of the backed up course. However, when you use CSV to do the process, the only dates rolling correctly are the week names. All other dates: restriction dates for viewing them, open, due, cut off for assignments, availability for forum ratings, due and close dates for forums, open and close phase dates for workshops, remain stuck in the original dates of the course used as template. 
In short, you can do one by one, you can't do it with a CSV file. So it is a bug that affects Moodle administrators trying to restore many courses for the new term.

But anyway, it is a moodle problem :)
